clients/src/main/java/org/apache/kafka/common/config/provider/FileConfigProvider.java

```diff
@@ -84,7 +112,17 @@ public ConfigData get(String path, Set<String> keys) {
         if (path == null || path.isEmpty()) {
             return new ConfigData(data);
         }
-        try (Reader reader = reader(path)) {
+
+        Path filePath = new File(path).toPath();
+        if (allowedPaths != null) {
+            long allowed = allowedPaths.stream().filter(allowedPath -> filePath.startsWith(allowedPath) || filePath.equals(allowedPath)).count();
```

Comment:
Do we need the `equals()` check? I think `startsWith()`  return` `true` is both paths are equals.